### DISCUSSION TITLE
defpoll not returning anything even though script works from command line

### DISCUSSION THREAD

**Discussion by violetcircus** [elkowar/eww#954] [Q&A] [answered]
hi im not sure if this is the best place for this but im stumped. I wrote a widget that gets information about the currently playing song on spotify using playerctl and had successfully implemented album covers but now (defpoll) doesn't seem to be running the script at all. Here's my defpoll statement:

```
(defpoll albumcover
:interval "1s"
"$HOME/.config/eww/scripts/album_cover.sh")
```

and where I use the variable (though it also returns nothing when I send it to a label):

```
(box :class "albumcover" :style 'backgound-image: url(\"${albumcover}\"); background-size: cover'
:width 300
:height 300)
```

the script works from the terminal using ./album_cover.sh in the scripts directory but returns nothing when run from a defpoll statement. I've checked and it's executable, the path to it is correct, and when run from the command line it spits out a perfectly useable path to the current album cover. I would appreciate any help i could get bc i'm totally out of ideas here

here's the bash script for those curious/ppl who may find it useful:

```
#!/bin/bash

PLAYER="spotify"

# Get the URL using playerctl
url=$(playerctl --player=$PLAYER --no-messages metadata | grep artUrl | awk '{ print $3 }' )
#check song status using playerctl
playerstatus=$(playerctl -s --player=$PLAYER --messages status --follow 2>/dev/null) 

if [ playerstatus == "No players found" ]; then
  echo "$HOME/.config/eww/src/img/nosong.png"
  exit 1
else
  # Check if the URL is empty
  if [ -z "$url" ]; then
    echo "$HOME/.config/eww/src/img/nosong.png"
    exit 1
  fi

  # Specify the directory where you want to save the downloaded file
  download_dir="$HOME/Pictures/music-thumbs/"

  # Use wget to download the file to the specified directory
  filename=$(basename "$url")
  output_file="$download_dir$filename.jpeg"

  if ! [ -f $output_file ]; then 
    wget -q "$url" -O "$output_file"

  # Check if wget was successful
    if [ $? -eq 0 ]; then
     echo "$output_file"
     exit 1
   else
      echo "$HOME/.config/eww/src/img/nosong.png"
      exit 1
    fi
  else
    echo "$output_file" 
    exit 1
  fi
fi
```

images for reference:

what it  looked like when it was working
![image](https://github.com/elkowar/eww/assets/26161381/6bc48b8f-6125-494c-8222-90d35ca021af)

what it looks like now
![image2](https://github.com/elkowar/eww/assets/26161381/7261be18-4006-44cb-914e-f2a46ae949a7)

a label with the :text "${albumcover}" property
![image3](https://github.com/elkowar/eww/assets/26161381/c3589ce8-97fa-4619-a6c0-b4865a087107)

let me know if any extra info would be helpful i'm pretty new to linux and stuff like this lol i'd never written any bash until like, yesterday

**Comment by violetcircus** [comment]
just tried to implement the battery widget from [rhyxn's bar](https://github.com/rxyhn/tokyo) and ran into the exact same issue. tried moving the defpoll to a different file to see if that was the issue and no dice. maybe a bug?

**Comment by w-lfchen** [comment, marked as answer]
I did two things in order to make your script work on my end:
1. Fix the typo in the box style: `backgound-image` -> `background-image`
2. Don't exit the script with exit code 1, exit with exit code 0 so just use `exit` instead of `exit 1`

**Reply by violetcircus** [reply]
oh my bad I messed up in my post there's no typo in the original file. changing the exit code in that one script seems to have fixed the issue with all my scripts not running though ty so much
